### DISCUSSION TITLE
SQL bridge to REST

### DISCUSSION THREAD

**Discussion by fredcwbr** [oguimbal/pg-server#4] [Q&A] [answered]
Howdy,

Would this emulator allow to create a bridge intercepting PG-SQL  SELECT (thats all I need ), and execute a REST post, interpreting the results as a table-like , from an ODBC perspective?

Ex., 
SELECT * 
FROM  myRESTserviceAPI.service 
WHERE field1='111111111' 
AND    field2='2222222'   

this would result in a lookup for the fake schema matching of REST APIs,  like , 

<table>
<tr><td colspan=3>RESTapiTABLE_LOOKUP</td></tr>
<tr><td> fake_schema_name </td><td> urlRESTAPI </td><td> eventualAuthInfo </td></tr>
<tr><td> myRESTserviceAPI </td><td> http://www.myrestapisite.wtvr/etc/ </td><td> someEncoded64AuthForExample </td></tr>
<tr><td> myRESTserviceAPI1 </td><td> http://www.myrestapisite1.wtvr/etc2/ </td><td> otherEncoded64AuthForExample </td></tr>
<tr><td> myRESTserviceAPIsoOn </td><td> http://www.myrestapisitesoOn.wtvr/more/etct/ </td><td> etcEncoded64AuthForExample </td></tr>
</table>

and would create a http_post like :
`headers= .... authinfo ... from lookup table`
`POST http://www.myrestapisite.wtvr/etc/service`
`{`
`field1=......  `
`field1=..... `
`......   from WHERE  CLAUSE., `
`}`

Would only consider for this matter,  WHERE clause as source of name/variable pairs to the API., 
,,
whence the JSONResult would be converted to a table-like 

from a JSON      [ {"fld1" : "a"   ,  "fld2" : "b",    "fldn" : "c"},  { jsonRec1 } , {jsonRec2}, ...... ] 
to a
<pre>
fld1     fld2    fldn
a          b         c   
d          e         f
.....

</pre>
Hoping to make myself clear, 

Thanks in advance

Best regards

from Brasil , 
Frederico C Wilhelms

**Comment by oguimbal** [comment, marked as answer]
Hi !

Yes, I guess that would be doable using `createAdvancedServer()` .

Ideally, that would need `createSimpleServer()` to be implemented more easily, but I dont have time to implement that this year :/
